rocksdb/src/main/java/com/yahoo/ycsb/db/rocksdb/RocksDBClient.java

```diff
@@ -0,0 +1,387 @@
+/*
+ * Copyright (c) 2017 YCSB contributors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package com.yahoo.ycsb.db.rocksdb;
+
+import com.yahoo.ycsb.*;
+import com.yahoo.ycsb.Status;
+import net.jcip.annotations.GuardedBy;
+import org.rocksdb.*;
+
+import java.io.*;
+import java.nio.ByteBuffer;
+import java.nio.file.*;
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+/**
+ * RocksDB binding for <a href="http://rocksdb.org/">RocksDB</a>.
+ *
+ * See {@code rocksdb/README.md} for details.
+ *
+ * @author <a href="mailto:adam@evolvedbinary.com">Adam Retter</a>
+ */
+public class RocksDBClient extends DB {
+
+  public static final String PROP_ROCKSDB_DIR = "rocksdb.dir";
+  private static final String CF_NAMES_FILENAME = "CF_NAMES";
+
+  @GuardedBy("RocksDBClient.class") private static Path rocksdbDir = null;
+  @GuardedBy("RocksDBClient.class") private static RocksObject dbOptions = null;
+  @GuardedBy("RocksDBClient.class") private static RocksDB rocksDb = null;
+  @GuardedBy("RocksDBClient.class") private static int references = 0;
+
+  private static final ConcurrentMap<String, ColumnFamily> COLUMN_FAMILIES = new ConcurrentHashMap<>();
+  private static final ConcurrentMap<String, Lock> COLUMN_FAMILY_LOCKS = new ConcurrentHashMap<>();
+
+  @Override
+  public void init() throws DBException {
+    synchronized(RocksDBClient.class) {
+      if(rocksDb == null) {
+        rocksdbDir = Paths.get(getProperties().getProperty(PROP_ROCKSDB_DIR));
+        System.out.println("RocksDB data dir: " + rocksdbDir);
+
+        try {
+          rocksDb = initRocksDB();
+        } catch (final IOException | RocksDBException e) {
+          throw new DBException(e);
+        }
+      }
+
+      references++;
+    }
+  }
+
+  private RocksDB initRocksDB() throws IOException, RocksDBException {
+    if(!Files.exists(rocksdbDir)) {
+      Files.createDirectories(rocksdbDir);
+    }
+
+    final List<String> cfNames = loadColumnFamilyNames();
+    final List<ColumnFamilyOptions> cfOptionss = new ArrayList<>();
```

Comment:
the `ss` looks like a typo. Is there no singular `ColumnFamilyOption`?

Reply:
The `ss` was intentional - to signify a list of plurals

Reply:
Should've been more explicit... I saw that it's intentional. That does not make it look less like a typo, honestly. Just saying that this could be a source of confusion

Reply:
Ah okay. Hmm... not exactly sure of how to improve that in a way that is satisfying. How about `allCfOptions`?